cider/tests/utils/DuckDbQueryRunner.cpp

```diff
@@ -388,6 +390,164 @@ void addColumnDataToCiderBatch<CiderByteArray>(
   }
 }
 
+template <typename T>
+void addColumnDataToScalarBatch(std::unique_ptr<duckdb::DataChunk>& dataChunk,
+                                int i,
+                                int row_num,
+                                CiderBatch* child) {
+  CHECK(child->resizeBatch(row_num, true));
+  auto data_buffer = child->asMutable<ScalarBatch<T>>()->getMutableRawData();
+  auto null_buffer = child->getMutableNulls();
+  int64_t null_count = 0;
+
+  for (int j = 0; j < row_num; j++) {
+    T value = std::numeric_limits<T>::min();  // init with Cider null value
+    if (dataChunk->GetValue(i, j).IsNull()) {
```

Comment:
I noticed that `DataChunk` has such method `ToArrowArray`, maybe we can use this.